node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+            garbage_largest_exceeding_balance_recently_qualified,
+        ));
+        let unconfirmed_adjustments = AdjustmentComputer::default()
+            .compute_unconfirmed_adjustments(weighted_accounts, cw_service_fee_balance_minor);
+        // The results are sorted from the biggest weights down
+        assert_eq!(
+            unconfirmed_adjustments[1].wallet(),
+            &wallet_of_expected_outweighed
+        );
+        // The weight of this account grew progressively due to the additional criterion added
+        // in to the sum. Consequences would've been that redistribution of the adjusted balances
+        // would've attributed this account with a larger amount to pay than it would've
+        // contained before the test started. To prevent that, we used to secure a rule that
+        // an account could never demand more than 100% of itself.
+        //
+        // Later it was changed to other policy. so called "outweighed" account gains automatically
+        // a balance equal to its disqualification limit, also a prominent front position in
+        // the resulting set of the accounts to pay out. Additionally, due to its favorable position,
+        // it can be given a bit more from the remains still languishing in the consuming wallet.
+        let proposed_adjusted_balance = unconfirmed_adjustments[1].proposed_adjusted_balance_minor;
+        assert!(
+            proposed_adjusted_balance > (original_balance_of_outweighed_account * 11 / 10),
+            "we expected the proposed balance at least 1.1 times bigger than the original balance \
+            which is {} but it was {}",
+            original_balance_of_outweighed_account.separate_with_commas(),
+            proposed_adjusted_balance.separate_with_commas()
+        );
+    }
+
+    #[test]
+    fn adjustment_started_but_all_accounts_were_eliminated_anyway() {
+        let test_name = "adjustment_started_but_all_accounts_were_eliminated_anyway";
+        let now = SystemTime::now();
+        let balance_1 = multiple_by_billion(3_000_000);
+        let account_1 = PayableAccount {
+            wallet: make_wallet("abc"),
+            balance_wei: balance_1,
+            last_paid_timestamp: now.checked_sub(Duration::from_secs(50_000)).unwrap(),
+            pending_payable_opt: None,
+        };
+        let balance_2 = multiple_by_billion(2_000_000);
+        let account_2 = PayableAccount {
+            wallet: make_wallet("def"),
+            balance_wei: balance_2,
+            last_paid_timestamp: now.checked_sub(Duration::from_secs(50_000)).unwrap(),
+            pending_payable_opt: None,
+        };
+        let balance_3 = multiple_by_billion(5_000_000);
+        let account_3 = PayableAccount {
+            wallet: make_wallet("ghi"),
+            balance_wei: balance_3,
+            last_paid_timestamp: now.checked_sub(Duration::from_secs(70_000)).unwrap(),
+            pending_payable_opt: None,
+        };
```

Comment:
It would be better if you explain the disqualification process using a comment.

The way you explained this test to me verbally was more explanatory than the current state of the test (along with the comments), so please add them.